[V3] --token and --no-instance flags

laggron42 opened the pull request:
### Type

- [ ] Bugfix
- [x] Enhancement
- [x] New feature

### Description of the changes

This PR brings two new flags to `redbot`:

- The `--token` flag that will run Red with the given token instead of the saved one.
- The `--no-instance` flag that will run Red without any instance. The data will be saved under a `tmp` dir and removed on next restart.

These flags can be combined for testing on Travis CI for example : 

```
python3 -m redbot --no-instance --prefix ! --token $MY_PREFIX
```
That command will run Red without any setup, useful for online automated testing !

laggron42 committed:
Ability to run Red with token without instance

laggron42 committed:
--no-instance flag

laggron42 committed:
Reformatted cli with black

Tobotimus commented on redbot/core/data_manager.py:
```diff
@@ -39,6 +41,23 @@
 config_file = config_dir / "config.json"
 
 
+def save_default_config():
```
I think this should be named `create_temp_config`, to be clearer about what it does.

Tobotimus commented on redbot/core/cli.py:
```diff
@@ -138,6 +138,16 @@ def parse_cli_flags(args):
+        help=(
+            "Run Red without any existing instance. "
+            "The data will be saved under a temporary folder "
+            "and deleted on next restart."
```
"deleted on next restart" could be interpreted as the next bot restart. I would specify "system restart" to be clearer.

Tobotimus commented on redbot/__main__.py:
```diff
@@ -106,9 +106,17 @@ def main():
+    if cli_flags.no_instance:
+        print(
+            "\033[1m"
+            "Warning: The data will be placed in a temporary folder and removed on next reboot."
```
Specify "system reboot" here too

Tobotimus committed:
Merge branch 'V3/develop' into token

laggron42 committed:
Fix changes requested by @Tobotimus

- Use "system reboot" to be clearer
- save_default_config renamed to create_temp_config
- More documentation for the create_temp_config function

Tobotimus commented on redbot/__main__.py:
```diff
@@ -106,9 +106,17 @@ def main():
+            "\033[0m"
+        )
+        cli_flags.instance_name = "temporary_red"
+        save_default_config()
```
`save_default_config` call needs to be updated

laggron42 committed:
Update create_temp_config call

laggron42 commented:
Fixed, sorry for that dumb error.

Tobotimus committed:
Merge branch 'V3/develop' into token

laggron42 committed:
Merge 755e9d577a9fe9c809cb5efbfcc16f805a1c8142 into c1bcca44328e995e6e9fb3bb28b6f10517c2a35d

Tobotimus committed:
Fix up imports
